### ISSUE TITLE
pyenv install on apple m3 fails with ar issue

### ISSUE THREAD

**Issue by step21** [pyenv/pyenv#2989]
The building seems to be working, but then ar complains at one module or another that it would be on a read-only file system. This is especially weird, as before it was unpacking / building like normal. Usually it stops at "libexpat". I'm not really sure if this is only a me-problem, or if it's packaging specific. It does however not seem packaging specific, as I also tried pyenv when not installed via brew, and the result was basically the same. And I couldn't find any similar issues or answers on other sites.

```
/opt/homebrew/bin/ginstall -c -m 644 ./Tools/gdb/libpython.py python.exe-gdb.py
clang -c -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall    -std=c11 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Werror=implicit-function-declaration -fvisibility=hidden  -I./Include/internal  -I. -I./Include -I/opt/homebrew/opt/ncurses/include -I/opt/homebrew/opt/readline/include -I/opt/homebrew/opt/readline/include -I/Users/utopist/.pyenv/versions/3.11.8/include -I/opt/homebrew/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include -I/opt/homebrew/opt/ncurses/include -I/opt/homebrew/opt/readline/include -I/opt/homebrew/opt/readline/include -I/Users/utopist/.pyenv/versions/3.11.8/include -I/opt/homebrew/include -I/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include  -DPy_BUILD_CORE -o Programs/_testembed.o ./Programs/_testembed.c
sed -e "s,@EXENAME@,/Users/utopist/.pyenv/versions/3.11.8/bin/python3.11," < ./Misc/python-config.in >python-config.py
LC_ALL=C sed -e 's,\$(\([A-Za-z0-9_]*\)),\$\{\1\},g' < Misc/python-config.sh >python-config
./Modules/expat/xmlparse.c:3116:9: warning: code will never be executed [-Wunreachable-code]
        parser->m_characterDataHandler(parser->m_handlerArg, parser->m_dataBuf,
        ^~~~~~
./Modules/expat/xmlparse.c:3115:16: note: silence by adding parentheses to mark code as explicitly dead
      else if (0 && parser->m_characterDataHandler)
               ^
               /* DISABLES CODE */ ( )
./Modules/expat/xmlparse.c:4059:9: warning: code will never be executed [-Wunreachable-code]
        parser->m_characterDataHandler(parser->m_handlerArg, parser->m_dataBuf,
        ^~~~~~
./Modules/expat/xmlparse.c:4058:16: note: silence by adding parentheses to mark code as explicitly dead
      else if (0 && parser->m_characterDataHandler)
               ^
               /* DISABLES CODE */ ( )
2 warnings generated.
rm -f Modules/expat/libexpat.a
ar rcs Modules/expat/libexpat.a Modules/expat/xmlparse.o Modules/expat/xmlrole.o Modules/expat/xmltok.o
ar: temporary file: Read-only file system
make: *** [Modules/expat/libexpat.a] Error 1
make: *** Waiting for unfinished jobs....
+++(/opt/homebrew/Cellar/pyenv/2.4.3/plugins/python-build/bin/python-build:837): build_package_standard_build(): build_failed

+++(/opt/homebrew/Cellar/pyenv/2.4.3/plugins/python-build/bin/python-build:148): build_failed(): echo
+++(/opt/homebrew/Cellar/pyenv/2.4.3/plugins/python-build/bin/python-build:149): build_failed(): colorize 1 'BUILD FAILED'
+++(/opt/homebrew/Cellar/pyenv/2.4.3/plugins/python-build/bin/python-build:99): colorize(): '[' -t 1 ']'
BUILD FAILED+++(/opt/homebrew/Cellar/pyenv/2.4.3/plugins/python-build/bin/python-build:99): colorize(): printf '\e[%sm%s\e[m' 1 'BUILD FAILED'
```

### Prerequisite
* [x] Make sure your problem is not listed in [the common build problems](https://github.com/pyenv/pyenv/wiki/Common-build-problems).
* [x] Make sure no duplicated issue has already been reported in [the pyenv issues](https://github.com/pyenv/pyenv/issues). You should look for closed issues, too.
* [ ] Make sure you are not asking us to help solving your specific issue.
  * GitHub issues is opened mainly for development purposes. If you want to ask someone to help solving your problem, go to some community site like [Gitter](https://gitter.im/yyuu/pyenv), [StackOverflow](https://stackoverflow.com/questions/tagged/pyenv), etc.
* [x] Make sure your problem is not derived from packaging (e.g. [Homebrew](https://brew.sh)).
  * Please refer to the package documentation for the installation issues, etc.
* [x] Make sure your problem is not derived from plugins.
  * This repository is maintaining `pyenv` and the default `python-build` plugin only. Please refrain from reporting issues of other plugins here.

### Description
- [x] Platform information (e.g. Ubuntu Linux 16.04): MacOS Sonoma 14.4
- [x] OS architecture (e.g. amd64): arm64
- [x] pyenv version: 2.4.3
- [x] Python version: 3.11.8
- [x] C Compiler information (e.g. gcc 7.3): clang 15.00
- [x] Please attach the debug trace of the failing command as a gist:
  * Run `env PYENV_DEBUG=1 <faulty command> 2>&1 | tee trace.log` and attach `trace.log`. E.g. if you have a problem with installing Python, run `env PYENV_DEBUG=1 pyenv install -v <version> 2>&1 | tee trace.log` (note the `-v` option to `pyenv install`).
- [x] If you have a problem with installing Python, please also attach `config.log` from the build directory
  * The build directory is reported after the "BUILD FAILED" message and is usually under `/tmp`.
- [ ] If the build succeeds but the problem is still with the build process (e.g. the resulting Python is missing a feature), please attach
  * the debug trace from reinstalling the faulty version with `env PYENV_DEBUG=1 pyenv install -f -k -v <version> 2>&1 | tee trace.log`
  * `config.log` from the build directory. When using `pyenv install` with `-k` as per above, the build directory will be under `$PYENV_ROOT/sources`.
  
```

```
[trace.log](https://gist.github.com/step21/1e1b0bf08348a7543fd5fc213e6403ef)
[config.log](https://gist.github.com/step21/be06c8679396196b128c023282a257c5)

**Comment by native-api** [comment]
Googling around, [I see that can be due to filesystem errors](https://askubuntu.com/questions/287021/how-to-fix-read-only-file-system-error-when-i-run-something-as-sudo-and-try-to).
FWIW, [`ar` uses TMPDIR for temporary files](https://www.unix.com/man-page/osx/1/ar/).

An antivirus is also a usual suspect, especially if your machine is "Managed", they reported

**Comment by step21** [comment]
> Googling around, [I see that can be due to filesystem errors](https://askubuntu.com/questions/287021/how-to-fix-read-only-file-system-error-when-i-run-something-as-sudo-and-try-to). FWIW, [`ar` uses TMPDIR for temporary files](https://www.unix.com/man-page/osx/1/ar/).
> 
> An antivirus is also a usual suspect, especially if your machine is "Managed".

Thanks for the reply. I think file system errors are very unlikely, as this machine is 1-2 weeks old and nothing else have this problem. The `TMPDIR` thing is more likely. There is no antivirus, apart from anything Apple ships by default.
It seems to work somewhat (at least manually) when explicitly setting `$TMPDIR`. This shouldn't be necessary as it should default to `/tmp` if unset. However apparently `ar` doesn't get that somehow?
Anyway, setting this manually worked. But then I also saw that I had unset it in `.bashrc` due to some tmux issue in the past. I outcommented this line for now, if it causes tmux problems I'll try to resolve it then.
Thx for th TMPDIR hint.
